### ISSUE TITLE
Appium is not identifying any object with xPath for a new page where as its working fine for rest of the pages

### ISSUE THREAD

**Issue by niranjanparija** [appium/java-client#328]
Appium is not identifying any object with xPath for a new page where as its working fine for rest of the pages with same xpath

For the new page,we can identify the Objects with Class.Can you please suggest any work out for this.
@TikhomirovSergey - Please suggest.

**Comment by SrinivasanTarget** [comment]
@niranjanparija Please update the logs as gist here and the code snippet that you are using..

**Comment by niranjanparija** [comment]
Script : 
lbl_company = new Label(device,"text='Company'");
lbl_title.verify();

Driver:
 String elemIdentifier1 = "//*[@" + elemIdentifier + "]";

        if (!driver.findElements(By.xpath(elemIdentifier1)).isEmpty()){

            try{
                isElementPresent = driver.findElementByXPath(elemIdentifier1).isDisplayed();
                Assert.assertTrue("Targeted element Loader is not present on screen",isElementPresent);
            } catch (Throwable t) {
                throw new DrvException(t.getMessage(), t);
            }
        }
Log:
 *** Loading page class com.test.automation.retail.android.pageobjects.generic.standingOrder.ChoosePayeeCompanySOPagePO *** 
Within Appium Driver verifyElementFound
text='Choose Payee'
text='Choose Payee'
12:43:02.008 [main] DEBUG c.t.automation.retail.RetailTestBase - Generating report.
12:43:02.009 [main] INFO  c.t.automation.retail.RetailTestBase - 
 ------------------------- ENDED TEST :com.test.automation.retail.android.scripts.nwb.sodd.NWB_CreateSO_1




org.openqa.selenium.WebDriverException: An unknown server-side error occurred while processing the command. (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 97 milliseconds
Build info: version: '2.48.2', revision: '41bccdd10cf2c0560f637404c2d96164b67d9d67', time: '2015-10-09 13:08:06'
System info: host: 'testingbjss-PC', ip: '192.168.237.66', os.name: 'Windows 7', os.arch: 'amd64', os.version: '6.1', java.version: '1.7.0_75'
*** Element info: {Using=xpath, value=//*[@text='Choose Payee']}
Session ID: 2992bc28-589e-4717-a2d2-232cf5484034
Driver info: io.appium.java_client.android.AndroidDriver
Capabilities [{platform=LINUX, app=C:\Temp\NWB_SERVER[SERVER_REF19_TEST_MONITISE_EU]_TYPE[RELEASE]_VERSION[04.06.0000]_BUILD[8].apk, javascriptEnabled=true, browserName=Android, networkConnectionEnabled=true, desired={app=C:\Temp\NWB_SERVER[SERVER_REF19_TEST_MONITISE_EU]_TYPE[RELEASE]_VERSION[04.06.0000]_BUILD[8].apk, deviceName=401710e5, platformName=Android, noReset=true, version=4.4.2}, locationContextEnabled=false, noReset=true, version=4.4.2, platformVersion=5.0.1, databaseEnabled=false, platformName=Android, deviceName=401710e5, webStorageEnabled=false, warnings={}, takesScreenshot=true}]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:206)
	at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:158)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:647)
	at io.appium.java_client.DefaultGenericMobileDriver.execute(DefaultGenericMobileDriver.java:42)
	at io.appium.java_client.AppiumDriver.execute(AppiumDriver.java:1)
	at io.appium.java_client.android.AndroidDriver.execute(AndroidDriver.java:1)
	at org.openqa.selenium.remote.RemoteWebDriver.findElements(RemoteWebDriver.java:378)
	at org.openqa.selenium.remote.RemoteWebDriver.findElementsByXPath(RemoteWebDriver.java:494)
	at io.appium.java_client.DefaultGenericMobileDriver.findElementsByXPath(DefaultGenericMobileDriver.java:136)
	at io.appium.java_client.AppiumDriver.findElementsByXPath(AppiumDriver.java:119)
	at org.openqa.selenium.By$ByXPath.findElements(By.java:356)
	at org.openqa.selenium.remote.RemoteWebDriver.findElements(RemoteWebDriver.java:341)
	at io.appium.java_client.DefaultGenericMobileDriver.findElements(DefaultGenericMobileDriver.java:47)
	at io.appium.java_client.AppiumDriver.findElements(AppiumDriver.java:86)
	at com.test.automation.framework.driver.AppiumDriver.verifyElementFound(AppiumDriver.java:619)
	at com.test.automation.framework.device.GenericDevice.verify(GenericDevice.java:321)
	at com.test.automation.framework.pageObject.pageElements.PageElement.verify(PageElement.java:68)
	at com.test.automation.retail.android.pageobjects.templates.standingOrder.ChoosePayeeCompanySO_Template.verifyPageElements(ChoosePayeeCompanySO_Template.java:46)
	at com.test.automation.retail.android.pageobjects.generic.standingOrder.ChoosePayeeCompanySOPagePO.verifyPage(ChoosePayeeCompanySOPagePO.java:34)
	at com.test.automation.retail.android.scripts.nwb.sodd.NWB_CreateSO.createSoPayeeName(NWB_CreateSO.java:69)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:78)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:212)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:68)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)


Process finished with exit code -1

**Comment by niranjanparija** [comment]
I think the issue is because we cannot retrieve the pagesource .Please find the console log for the same page where as I can get the page source for rest of the pages where xpath is working fine.

 *** Loading page class com.test.automation.retail.android.pageobjects.generic.standingOrder.ChoosePayeeCompanySOPagePO *** 
Within Appium Driver getElementTextWithIndex elemIdentifier : android.widget.TextView
element found by class with index
driver.getPageSource()
driver.getPageSource()
driver.getPageSource()
driver.getPageSource()
13:06:22.951 [main] DEBUG c.t.automation.retail.RetailTestBase - Generating report.
13:06:22.952 [main] INFO  c.t.automation.retail.RetailTestBase - 
 ------------------------- ENDED TEST :com.test.automation.retail.android.scripts.nwb.sodd.NWB_CreateSO_1




org.openqa.selenium.WebDriverException: An unknown server-side error occurred while processing the command. (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 157 milliseconds
Build info: version: '2.48.2', revision: '41bccdd10cf2c0560f637404c2d96164b67d9d67', time: '2015-10-09 13:08:06'
System info: host: 'testingbjss-PC', ip: '192.168.237.66', os.name: 'Windows 7', os.arch: 'amd64', os.version: '6.1', java.version: '1.7.0_75'
Session ID: 5af27f65-c138-4d79-aafa-b6e7536507e3
Driver info: io.appium.java_client.android.AndroidDriver
Capabilities [{platform=LINUX, app=C:\Temp\NWB_SERVER[SERVER_REF19_TEST_MONITISE_EU]_TYPE[RELEASE]_VERSION[04.06.0000]_BUILD[8].apk, javascriptEnabled=true, browserName=Android, networkConnectionEnabled=true, desired={app=C:\Temp\NWB_SERVER[SERVER_REF19_TEST_MONITISE_EU]_TYPE[RELEASE]_VERSION[04.06.0000]_BUILD[8].apk, deviceName=401710e5, platformName=Android, noReset=true, version=4.4.2}, locationContextEnabled=false, noReset=true, version=4.4.2, platformVersion=5.0.1, databaseEnabled=false, platformName=Android, deviceName=401710e5, webStorageEnabled=false, warnings={}, takesScreenshot=true}]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:206)
	at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:158)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:647)
	at io.appium.java_client.DefaultGenericMobileDriver.execute(DefaultGenericMobileDriver.java:42)
	at io.appium.java_client.AppiumDriver.execute(AppiumDriver.java:206)
	at org.openqa.selenium.remote.RemoteWebDriver.getPageSource(RemoteWebDriver.java:501)
	at com.test.automation.framework.driver.AppiumDriver.getElementTextWithIndex(AppiumDriver.java:481)
	at com.test.automation.framework.device.AndroidDevice.getElementTextWithIndex(AndroidDevice.java:59)
	at com.test.automation.retail.android.pageobjects.templates.standingOrder.ChoosePayeeCompanySO_Template.verifyPageElements(ChoosePayeeCompanySO_Template.java:44)
	at com.test.automation.retail.android.pageobjects.generic.standingOrder.ChoosePayeeCompanySOPagePO.verifyPage(ChoosePayeeCompanySOPagePO.java:34)
	at com.test.automation.retail.android.scripts.nwb.sodd.NWB_CreateSO.createSoPayeeName(NWB_CreateSO.java:69)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:78)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:212)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:68)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)


Process finished with exit code -1

**Comment by niranjanparija** [comment]
Can anyone please suggest any solution.

**Comment by niranjanparija** [comment]
Is it because of any compressed or Uncompressed hierarchy? If yes,is there any way out?

**Comment by niranjanparija** [comment]
@bootstraponline @jlipps @Jonahss

**Comment by TikhomirovSergey** [comment]
@niranjanparija 
I can't understand what happens...

Does it happen when you try to get page source? 

```java
driver.getPageSource();
```

Could you provide code samples and server output as links to gists? 
https://help.github.com/articles/creating-gists/

Which server version do you use?

I think it may be the server-side problem...

**Comment by niranjanparija** [comment]
Appium Version is 1.4.13.

Regarding the issue,when I discussed with Developers of the app,I got to know that they used Page Viewer for this new page in the App where we are unable to identify Objects with Xpath (when I debugged to know the reason of the same,I noticed that we are not getting the PageSource for this specific Page with Driver.getPageSource also,which I assume is a problem and may be for the same we are facing issue in Identification of the Objects).

Basically the issue is We cannot identify any objects with Xpath for this new page,designed with PageViewer.

Please suggest.I have already pasted all the logs earlier in this issue.Please have a look at that,

**Comment by TikhomirovSergey** [comment]
@niranjanparija 
It is impossible to reproduce the problem. Please try it with appium node 1.5. 
If the problem is still here, please open a new one. Please try to format a new issue like this template: https://github.com/appium/java-client/blob/master/ISSUE_TEMPLATE.md

It is most likely the server-side problem.
